### PR TITLE
fix(statemachine)!: use path within IBC store without escaping characters in solomachine

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

As discussed with @colin-axner, I implemented changes so that:

- the bytes the solo machine signs over are in a key that does not include the IBC store prefix, since the IBC store prefix only makes send in a multistore, and the diversifier already satisfy the function of namespacing the data.
- both `String` and `Pretty` functions become unused and can be removed later on. They make assumptions about formatting a merkle path in a multi store into a string that we shouldn't be making.

ref: #4128 

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/23-commitment/types/merkle.go
Comment (crodriguezvega) [inline modules/core/23-commitment/types/merkle.go:85]: I added a deprecated notice to these functions instead of removing them completely because I think removing them is API breaking and if I remove them in this PR the backport to v7.3.x will be more difficult. I will remove them in a follow up PR once this one gets merged.

File: modules/core/exported/commitment.go
Comment (crodriguezvega) [inline modules/core/exported/commitment.go:31]: If we remove this function, I am not sure if this interface makes much sense anymore...

File: testing/solomachine.go
Comment (crodriguezvega) [inline testing/solomachine.go:519]: This testing functions in this file return a merle path with 2 elements (first one for the IBC store), I could also change those to return only the second element. So something like this for `GetClientStatePath`:

```
func (solo *Solomachine) GetClientStatePath(counterpartyClientIdentifier string) commitmenttypes.MerklePath {
  return commitmenttypes.NewMerklePath(host.FullClientStatePath(counterpartyClientIdentifier))
}
```

Although the function the becomes a one-liner, so maybe we don't even to have a separate function to construct the path. Then in the test I would get key 0, instead of 1 as the path over which the solo machine signs the bytes.

**Review by colin-axner — APPROVED** [review]
LGTM

File: testing/solomachine.go
Comment (crodriguezvega) [reply testing/solomachine.go:519]: Yes, I will do that better in a follow-up PR, otherwise if I remove these functions now that's also API breaking.

File: modules/light-clients/06-solomachine/client_state.go
Comment (damiannolan) [inline modules/light-clients/06-solomachine/client_state.go:136]: `GetKey` seems to do some unescaping on the string also, is this okay here?

from `GetKey`:

```
key, err := url.PathUnescape(mp.KeyPath[i])
```

**Review by damiannolan — APPROVED** [review]
Looks good to me! LGTM, left one question about the `GetKey` method

File: modules/light-clients/06-solomachine/client_state.go
Comment (colin-axner) [reply modules/light-clients/06-solomachine/client_state.go:136]: Maybe we should just use `merklePath.KeyPath[1]`?

Comment (DimitrisJim) [reply modules/light-clients/06-solomachine/client_state.go:136]: wouldn't this be fine considering escaping calls have been removed? It seems like it would just be best to tweak `GetKey`.

Comment (crodriguezvega) [reply modules/light-clients/06-solomachine/client_state.go:136]: That's a good observation @damiannolan! 

I think it is still fine, since it is doing unescaping. And also according to ICS 24 `%` is not a valid character in identifiers and paths, so it should not happen that a path contains an escaped character and then `GetKey` would return it unescaped.

Comment (crodriguezvega) [reply modules/light-clients/06-solomachine/client_state.go:136]: What do you guys prefer to do?
-

Comment (damiannolan) [reply modules/light-clients/06-solomachine/client_state.go:136]: Happy with opt 2 or 3 I think!

Comment (DimitrisJim) [reply modules/light-clients/06-solomachine/client_state.go:136]: if we're having a vote I'd go towards 3.

**Review by DimitrisJim — APPROVED** [review]

File: modules/light-clients/06-solomachine/client_state.go
Comment (crodriguezvega) [reply modules/light-clients/06-solomachine/client_state.go:136]: Ok, I will go for 2 and implement 3 in the next PR.

Comment (crodriguezvega) [reply modules/light-clients/06-solomachine/client_state.go:136]: Done. I will merge when tests pass.

**Pull request merged**
